src/Filesystem/Filesystem.php

```diff
@@ -33,6 +35,40 @@ public function add(string $file, Media $media, ?string $targetFileName = null)
         app(FileManipulator::class)->createDerivedFiles($media);
     }
 
+    public function addRemote(RemoteFile $file, Media $media, ?string $targetFileName = null)
+    {
+        $this->copyToMediaLibraryFromRemote($file, $media, null, $targetFileName);
+
+        event(new MediaHasBeenAdded($media));
+
+        app(FileManipulator::class)->createDerivedFiles($media);
+    }
+
+    public function copyToMediaLibraryFromRemote(RemoteFile $file, Media $media, ?string $type = null, ?string $targetFileName = null)
+    {
+        $storage = Storage::disk($file->getDisk());
+
+        $destinationFileName = $targetFileName ?: $file->getFilename();
+
+        $destination = $this->getMediaDirectory($media, $type).$destinationFileName;
+
+        if ($file->getDisk() === $media->getDiskDriverName()) {
+            $storage->copy($file->getKey(), $destination);
```

Comment:
If the file on the disk has visibility set to private, then the copied file will also be private. Should the copied file not be public by default?

Reply:
I'm not sure - probably a question for @freekmurze